docs/api-reference-widgets.md

```diff
@@ -0,0 +1,175 @@
+  * by default, a regular `input[type=text]` element is used.
+
+##### String formats
+
+The built-in string field also supports the JSONSchema `format` property, and will render an appropriate widget by default for the following string formats:
+
+- `email`: An `input[type=email]` element is used;
+- `uri`: An `input[type=url]` element is used;
+- `data-url`: By default, an `input[type=file]` element is used; in case the string is part of an array, multiple files will be handled automatically (see [File widgets](#file-widgets)).
+- `date`: By default, an `input[type=date]` element is used;
+- `date-time`: By default, an `input[type=datetime-local]` element is used.
+
+![](https://i.imgur.com/xqu6Lcp.png)
+
+Please note that, even though they are standardized, `datetime-local` and `date` input elements are not yet supported by Firefox and IE. If you plan on targeting these platforms, two alternative widgets are available:
+
+- `alt-datetime`: Six `select` elements are used to select the year, the month, the day, the hour, the minute and the second;
+- `alt-date`: Three `select` elements are used to select the year, month and the day.
+
+> **Firefox 57 - 66**: Firefox partially supporting `date` and `time` input types, but not `datetime-local`, `month` or `week`
+
+![](https://i.imgur.com/VF5tY60.png)
+
+You can customize the list of years displayed in the `year` dropdown by providing a ``yearsRange`` property to ``ui:options`` in your uiSchema. Its also possible to remove the `Now` and `Clear` buttons with the `hideNowButton` and `hideClearButton` options.
+
+```jsx
+uiSchema: {
+  a_date: {
+    "alt-datetime": {
+      "ui:widget": "alt-datetime",
+      "ui:options": {
+        yearsRange: [1980, 2030],
+        hideNowButton: true,
+        hideClearButton: true,
+      },
+    },
+  },
+},
+```
+
+#### For `number` and `integer` fields
+
+  * `updown`: an `input[type=number]` updown selector;
+  * `range`: an `input[type=range]` slider;
+  * `radio`: a radio button group with enum values. This can only be used when `enum` values are specified for this input.
+  * By default, a regular `input[type=text]` element is used.
+
+> Note: If JSONSchema's `minimum`, `maximum` and `multipleOf` values are defined, the `min`, `max` and `step` input attributes values will take those values.
+
+
+## Hidden widgets
+
+It's possible to use a hidden widget for a field by setting its `ui:widget` uiSchema directive to `hidden`:
+
+```js
+const schema = {
+  type: "object",
+  properties: {
+    foo: {type: "boolean"}
+  }
+};
+
+const uiSchema = {
+  foo: {"ui:widget": "hidden"}
+};
+```
+
+Notes:
+
+ - Hiding widgets is only supported for `boolean`, `string`, `number` and `integer` schema types;
+ - A hidden widget takes its value from the `formData` prop.
+
+## File widgets
+
+This library supports a limited form of `input[type=file]` widgets, in the sense that it will propagate file contents to form data state as [data-url](http://dataurl.net/#about)s.
```

Comment:
```suggestion
This library supports a limited form of `input[type=file]` widgets, in the sense that it will propagate file contents to form data state as [data-url](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs)s.
```